Denis_Protsenko.py

```diff
@@ -0,0 +1,218 @@
+from pydantic import BaseModel
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from typing import Union
+
+
+# Абстрактний клас
+class PublicModel(ABC):
+    @abstractmethod
+    def book_info(self):
+        pass
+
+
+# Клас моделі книги
+class BookModel(BaseModel):
+    title: str
+    author: str
+    year: int
+
+
+# Клас моделі журналу
+class MagazineModel(BaseModel):
+    title: str
+    author: str
+    year: int
+
+
+# Клас книги
+class Book(PublicModel):
+    def __init__(self, book_model: BookModel):
+        self.title = book_model.title
+        self.author = book_model.author
+        self.year = book_model.year
```

Comment:
it's fine to define separate attributes, but you can just do like
self.model = book.model
and in other places get attributes like
self.model.title